homeassistant/data_entry_flow.py

```diff
@@ -346,24 +369,24 @@ async def _async_handle_step(
             step_done.set_result(None)
 
         if result["type"] not in (
```

Comment:
We can just try to parse it here? `FlowResultType(result["type"])` will raise if it's invalid?

Reply:
Doh! It didn't register with me that it was enumrating all possibilities here.
It's a bank holiday here this week-end so might not get a chance to amend until Tuesday.

Reply:
I have adjusted this to add a warning if it's not already a `FlowResultType`:
- if it parses a correct value, it will display a warning.
- if it fails to parse, it will raise a ValueError.